manifests/agent/facter.pp

```diff
@@ -0,0 +1,69 @@
+# Puppet agent facter configuration
+# @api private
+class puppet::agent::facter inherits puppet::config {
+  puppet::config::agent {
+    'blocklist': value => $puppet::facter_blocklist;
+    'cachelist': value => $puppet::facter_cachelist;
+    'cache_ttl': value => $puppet::cache_ttl;
+  }
+
+    if versioncmp(fact('aio_agent_version'),'7') >= 0 {
+    file { '/etc/puppetlabs/facter':
+      ensure => directory,
+    }
+
+        Hocon_setting {
+      path    => '/etc/puppetlabs/facter/facter.conf',
+      require => File['/etc/puppetlabs/facter'],
+    }
+
+
+        if $blocklist {
+      hocon_setting { 'blocklist facts group':
+        ensure  => present,
+        setting => 'fact-groups.blocked-facts',
+        value   => $blocklist,
```

Comment:
there's no `$blocklist` in this class. you probably meant `$puppet::facter_blocklist`?